### DISCUSSION TITLE
Creating a Span using traceId

### DISCUSSION THREAD

**Discussion by pjjohnston** [open-telemetry/opentelemetry-java#3102] [Q&A]
Hi,

I have a scenario where I'm forced to propagate merely using a traceId (a streamer with multiple input topics))

I see that this seems possible using the static method:

SpanContext.create(String traceIdHex, String spanIdHex.........)

I'm assuming that I can generate the SpanId myself? Any pointers on this approach would be greatly appreciated.

**Comment by jkwatson** [comment]
Yes, you can definitely generate your own span ids. I'd take a look at what our `io.opentelemetry.sdk.trace.RandomIdGenerator` does for this.  `SpanId.fromLong(long)` will be your friend.

**Reply by pjjohnston** [reply]
Excellent - thanks John!
